java-frontend/src/main/java/org/sonar/java/model/JMethodSymbol.java

```diff
@@ -96,45 +99,81 @@ public TypeSymbol returnType() {
   @Nullable
   @Override
   public MethodSymbol overriddenSymbol() {
-    if (overriddenSymbol == Symbols.unknownMethodSymbol) {
-      overriddenSymbol = convertOverriddenSymbol();
-    }
-    return overriddenSymbol;
+    return overriddenSymbols().stream()
+      .findFirst()
+      .orElse(null);
   }
 
-  @Nullable
-  private MethodSymbol convertOverriddenSymbol() {
-    IMethodBinding overrides = find(
-      methodBinding()::overrides,
-      methodBinding().getDeclaringClass()
-    );
-    if (overrides == null) {
-      return null;
+  @Override
+  public List<MethodSymbol> overriddenSymbols() {
+    if (overriddenSymbols == null) {
+      overriddenSymbols = convertOverriddenSymbols();
     }
-    return sema.methodSymbol(overrides);
+    return overriddenSymbols;
   }
 
-  @Nullable
-  private IMethodBinding find(Predicate<IMethodBinding> predicate, ITypeBinding t) {
-    for (IMethodBinding candidate : t.getDeclaredMethods()) {
-      if (predicate.test(candidate)) {
-        return candidate;
-      }
-    }
-    for (ITypeBinding i : t.getInterfaces()) {
-      IMethodBinding r = find(predicate, i);
-      if (r != null) {
-        return r;
+  private List<MethodSymbol> convertOverriddenSymbols() {
+    IMethodBinding methodBinding = methodBinding();
+    return find(methodBinding::overrides, methodBinding.getDeclaringClass()).stream()
+      .map(sema::methodSymbol)
+      .collect(Collectors.toList());
+  }
+
+  private List<IMethodBinding> find(Predicate<IMethodBinding> predicate, ITypeBinding typeBinding) {
+    List<IMethodBinding> bindings = new ArrayList<>();
```

Comment:
If it's a list, we will hit the problem of duplicates, and so you have to explore it all the time to check if it's not already there. having a LinkedHashSet preserve the order of introduction in the collection, and will guarantee unicity already. I feel it might be easier.